app/iot/async_helper.py

```diff
@@ -0,0 +1,11 @@
+import asyncio
+from typing import Any, Awaitable
+
+
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> list:
+    return await asyncio.gather(*functions)
```

Comment:
The return type of `run_parallel` is specified as `list`, but `asyncio.gather` returns a list of results only if all the awaitables complete successfully. If any of the awaitables raise an exception, `asyncio.gather` will raise an exception and not return a list. Consider updating the return type to `list[Any]` or handling exceptions if you want to ensure a list is always returned.